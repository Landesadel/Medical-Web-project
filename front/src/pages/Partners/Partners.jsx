import React from 'react';
import {Link, useNavigate} from "react-router-dom";
import Virtumed from '../../assets/images/partners/Virtumed.png';
import Eidos from '../../assets/images/partners/Eidos.png';
import Impresa from '../../assets/images/partners/Impresa.png';
import styles from "./Partners.module.scss";
import Instagram from "../../assets/images/instagram.png";

const partners=[
    {
        preview:Virtumed,
        alt:"Virtumed",
        link:"https://virtumed.ru/"
    },
    {
        preview:Eidos,
        alt:"Eidos",
        link:"https://eidos-medicine.com/"
    },
    {
        preview:Impresa,
        alt:"Impresa",
        link:"https://impresaltd.com/"
    },
];

const Partners = () => {
    const navigate = useNavigate();
    return (
        <div className="container">
            <h1 className={styles.heading}>{'Компании-партнеры'}</h1>
            <div className={styles.tree}>
                <ul>
                    {partners?.map((item,idx)=>(
                        <li key={idx}>
                            <div className={styles.partnerWrapper}>
                                <div className={styles.previewWrapper}><img src={item?.preview} alt={item?.alt}/></div>
                                <Link
                                    to={item?.link}
                                    rel="nofollow noopener noreferrer"
                                    target="_blank"
                                    className="everywhere-button"
                                >
                                  <span>{item?.link}</span>
                                </Link>
                            </div>
                        </li>
                    ))}

                </ul>
            </div>
        </div>
    );
};

export default Partners;