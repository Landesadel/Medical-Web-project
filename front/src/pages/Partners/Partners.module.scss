@import '../../assets/styles/variables';

$max-tablet-width: '991px';
$max-mobile-width: '767px';
$max-mobile-sm-width: '545px';
$max-small-screen-width: '249px';

@mixin tablet {
  @media(max-width: $max-tablet-width) {
    @content;
  }
}

@mixin mobile {
  @media(max-width: $max-mobile-width) {
    @content;
  }
}

@mixin mobile-sm-screen {
  @media(max-width: $max-mobile-sm-width) {
    @content;
  }
}

@mixin small_screen {
  @media(max-width: $max-small-screen-width) {
    @content;
  }
}
.heading{
  text-align: center;
  margin: 20px 0 30px;
}
.tree {
  position: relative;
  display: flex;
  flex-direction: column;
  padding: 0 0 20px 0;
}
.partnerWrapper{
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 20px;
  background: #fff;
  padding: 20px 30px;
  box-shadow: inset 0 0 3px #9a9a9a;
  border-radius: 10px;
  @include tablet(){
    padding: 20px 15px;
  }
  @include mobile(){
    flex-direction: column;
    padding: 20px 10px;
   }
  
  a{
    flex: 1 1 auto;
    font-size: 26px;
    text-align: right;

    @include tablet(){
      font-size: 20px;
    }
    @include mobile(){
      text-align: center;
    }
  }

}
.previewWrapper{
  flex:0 1 auto;
  @include mobile(){
    margin-bottom: 20px;
  }
}