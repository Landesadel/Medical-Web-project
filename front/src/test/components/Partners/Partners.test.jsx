import React from 'react';
import Partners from "../../../pages/Partners/Partners";
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import {BrowserRouter, Routes} from 'react-router-dom';



describe('Partners', () => {
    it('render partners', () => {
        render(
            <BrowserRouter>
                <Partners/>
            </BrowserRouter>
        );

        expect(screen.getByRole('list')).toBeInTheDocument();
        expect(screen.getByText('https://virtumed.ru/')).toBeInTheDocument();
    });

    it('snapshot partners', () => {
        const snapshot=render(
            <BrowserRouter>
                <Partners/>
            </BrowserRouter>
        );
        expect(snapshot).toMatchSnapshot();

    });

});
