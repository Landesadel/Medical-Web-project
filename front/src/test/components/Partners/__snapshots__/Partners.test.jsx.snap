// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partners snapshot partners 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1
          class="heading"
        >
          Компании-партнеры
        </h1>
        <div
          class="tree"
        >
          <ul>
            <li>
              <div
                class="partnerWrapper"
              >
                <div
                  class="previewWrapper"
                >
                  <img
                    alt="Virtumed"
                    src=""
                  />
                </div>
                <a
                  class="everywhere-button"
                  href="https://virtumed.ru/"
                  rel="nofollow noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    https://virtumed.ru/
                  </span>
                </a>
              </div>
            </li>
            <li>
              <div
                class="partnerWrapper"
              >
                <div
                  class="previewWrapper"
                >
                  <img
                    alt="Eidos"
                    src=""
                  />
                </div>
                <a
                  class="everywhere-button"
                  href="https://eidos-medicine.com/"
                  rel="nofollow noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    https://eidos-medicine.com/
                  </span>
                </a>
              </div>
            </li>
            <li>
              <div
                class="partnerWrapper"
              >
                <div
                  class="previewWrapper"
                >
                  <img
                    alt="Impresa"
                    src=""
                  />
                </div>
                <a
                  class="everywhere-button"
                  href="https://impresaltd.com/"
                  rel="nofollow noopener noreferrer"
                  target="_blank"
                >
                  <span>
                    https://impresaltd.com/
                  </span>
                </a>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1
        class="heading"
      >
        Компании-партнеры
      </h1>
      <div
        class="tree"
      >
        <ul>
          <li>
            <div
              class="partnerWrapper"
            >
              <div
                class="previewWrapper"
              >
                <img
                  alt="Virtumed"
                  src=""
                />
              </div>
              <a
                class="everywhere-button"
                href="https://virtumed.ru/"
                rel="nofollow noopener noreferrer"
                target="_blank"
              >
                <span>
                  https://virtumed.ru/
                </span>
              </a>
            </div>
          </li>
          <li>
            <div
              class="partnerWrapper"
            >
              <div
                class="previewWrapper"
              >
                <img
                  alt="Eidos"
                  src=""
                />
              </div>
              <a
                class="everywhere-button"
                href="https://eidos-medicine.com/"
                rel="nofollow noopener noreferrer"
                target="_blank"
              >
                <span>
                  https://eidos-medicine.com/
                </span>
              </a>
            </div>
          </li>
          <li>
            <div
              class="partnerWrapper"
            >
              <div
                class="previewWrapper"
              >
                <img
                  alt="Impresa"
                  src=""
                />
              </div>
              <a
                class="everywhere-button"
                href="https://impresaltd.com/"
                rel="nofollow noopener noreferrer"
                target="_blank"
              >
                <span>
                  https://impresaltd.com/
                </span>
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
